queuebacca-core/src/main/java/io/axonif/queuebacca/subscribing/Subscriber.java

```diff
@@ -52,17 +63,32 @@
     private final WorkExecutorFactory workExecutorFactory;
     private final ExceptionResolver exceptionResolver;
 
+	private final int finalFallbackReadCount;
+    private final FallbackMessageConsumerFactory finalFallbackMessageConsumerFactory;
+
 	/**
 	 * Creates a new instance of a {@link Subscriber} for a specific {@link Client}.
 	 *
 	 * @param client the client for the message broker
 	 * @param workExecutorFactory factory for creating {@link WorkExecutor WorkExecutors}
 	 * @param exceptionResolver determines resolution for exceptions thrown by messages
 	 */
-	public Subscriber(Client client, WorkExecutorFactory workExecutorFactory, ExceptionResolver exceptionResolver) {
-        this.client = requireNonNull(client);
-		this.workExecutorFactory = requireNonNull(workExecutorFactory);
-		this.exceptionResolver = requireNonNull(exceptionResolver);
+	private Subscriber(
```

Comment:
`requireNonNull` on the constructor args? Or is the thinking that since this is private you can already decide guarantees about nullability?

Reply:
Yea that's the idea. I usually only do it if it's strictly necessary and hasn't been vetted somewhere else within the same context.